src/frameworks/index.ts

```diff
@@ -398,136 +389,225 @@ export async function prepareFrameworks(
+        // TODO move to templates
+        if (packageJson.type === "module") {
+          await writeFile(
+            join(functionsDist, "server.js"),
+            `import { onRequest } from 'firebase-functions/v2/https';
   const server = import('firebase-frameworks');
   export const ${functionId} = onRequest(${JSON.stringify(
-            frameworksBackend || {}
-          )}, (req, res) => server.then(it => it.handle(req, res)));
+              frameworksBackend || {}
+            )}, (req, res) => server.then(it => it.handle(req, res)));
   `
-        );
-      } else {
-        await writeFile(
-          join(functionsDist, "server.js"),
-          `const { onRequest } = require('firebase-functions/v2/https');
+          );
+        } else {
+          await writeFile(
+            join(functionsDist, "server.js"),
+            `const { onRequest } = require('firebase-functions/v2/https');
   const server = import('firebase-frameworks');
   exports.${functionId} = onRequest(${JSON.stringify(
-            frameworksBackend || {}
-          )}, (req, res) => server.then(it => it.handle(req, res)));
+              frameworksBackend || {}
+            )}, (req, res) => server.then(it => it.handle(req, res)));
   `
+          );
+        }
+
+        ignore = ["node_modules", ".git", "firebase-debug.log", "firebase-debug.*.log"];
+      } else {
+        const { requirementsTxt, imports } = codegenFunctionsResult;
+
+        await writeFile(
+          join(functionsDist, "requirements.txt"),
+          `${requirementsTxt.trim()}\nfirebase_functions`.trimStart()
+        );
+        await writeFile(
+          join(functionsDist, "main.py"),
+          `from firebase_functions import https_fn
+from io import BytesIO
+from ${imports![0]} import ${imports![1]} as discoveredApp
+def handle(app, environ):
+    status = None
+    headers = None
+    body = BytesIO()
+    
+    def start_response(rstatus, rheaders):
+        nonlocal status, headers
+        status, headers = rstatus, rheaders
+        
+    app_iter = app(environ, start_response)
+    try:
+        for data in app_iter:
+            assert status is not None and headers is not None, \\
+                "start_response() was not called"
+            body.write(data)
+    finally:
+        if hasattr(app_iter, 'close'):
+            app_iter.close()
+    return status, headers, body.getvalue()
+
+@https_fn.on_request()
+def ${functionId}(req: https_fn.Request) -> https_fn.Response:
+    status, headers, body = handle(discoveredApp, req.environ)
+    return https_fn.Response(body, status, headers)
+`
         );
+        const python = getProjectPath("venv", "bin", "python");
+        if (!dirExistsSync(join(functionsDist, "venv"))) {
+          spawnSync(python, ["-m", "venv", "venv"], { cwd: functionsDist });
+        }
+        await new Promise<void>((resolve) => {
+          const child = runWithVirtualEnv(
+            ["pip", "install", "-r", "requirements.txt"],
```

Comment:
I think `pip` could also run from `venv/bin/pip`. My project was erroring out because my `pip` alias was `pip3` instead of `pip`